indra/llcommon/llevents.h

```diff
@@ -137,6 +138,10 @@ typedef boost::signals2::signal<bool(const LLSD&), LLStopWhenHandled, float>  LL
 /// Methods that forward listeners (e.g. constructed with
 /// <tt>boost::bind()</tt>) should accept (const LLEventListener&)
 typedef LLStandardSignal::slot_type LLEventListener;
+/// Support a listener accepting (const LLBoundListener&, const LLSD&).
+/// Note that LLBoundListener::disconnect() is a const method: this feature is
+/// specifically intended to allow a listener to disconnect itself when done.
+typedef LLStandardSignal::extended_slot_type LLAwareListener;
```

Comment:
I feel like name 'aware' might be misleading for a self terminating listener. May be LLTemporaryListener?

Reply:
Mmm, Temporary sounds to me too much as though we're somehow constraining its lifespan.

What I want to convey with the name is the extended signature. `LLEnrichedListener`? `LLExtendedListener`?

Reply:
`LLSelfReferentialListener`? `LLReflexiveListener`? (not really better than the above I think)

Reply:
`LLAwareListener` is fine enough for me.